pylint/checkers/similar.py

```diff
@@ -43,15 +69,294 @@
+
+class LinesChunk:
+    """
+    The LinesChunk object computes and stores the hash of some consecutive stripped lines of a lineset.
+    """
+
+    __slots__ = (
+        "_fileid",
+        "_index",
+        "_hash",
+    )
+
+    def __init__(self, fileid: str, num_line: int, *lines: Iterable[str]) -> None:
+        self._fileid: str = fileid
+        """The name of the file from which the LinesChunk object is generated """
+
+        self._index: Index = Index(num_line)
+        """The index in the stripped lines that is the starting of consecutive lines"""
+
+        self._hash: int = sum(hash(lin) for lin in lines)
+        """The hash of some consecutive lines"""
+
+    def __eq__(self, o: Any) -> bool:
+        if not isinstance(o, LinesChunk):
+            return NotImplemented
+        return self._hash == o._hash
+
+    def __hash__(self) -> int:
+        return self._hash
+
+    def __repr__(self) -> str:
+        return (
+            f"<LinesChunk object for file {self._fileid} ({self._index}, {self._hash})>"
+        )
+
+    def __str__(self) -> str:
+        return (
+            f"LinesChunk object for file {self._fileid}, starting at line {self._index} \n"
+            f"Hash is {self._hash}"
+        )
+
+
+class SuccessiveLinesLimits:
+    """
+    A class to handle the numbering of begin and end of successive lines.
+
+    :note: Only the end line number can be updated.
+    """
+
+    __slots__ = ("_start", "_end")
+
+    def __init__(self, start: LineNumber, end: LineNumber) -> None:
+        self._start: LineNumber = start
+        self._end: LineNumber = end
+
+    @property
+    def start(self) -> LineNumber:
+        return self._start
+
+    @property
+    def end(self) -> LineNumber:
+        return self._end
+
+    @end.setter
+    def end(self, value: LineNumber) -> None:
+        self._end = value
+
+    def __repr__(self) -> str:
+        return f"<SuccessiveLinesLimits <{self._start};{self._end}>>"
+
+
+class LineSetStartCouple(NamedTuple):
+    """
+    Indices in both linesets that mark the beginning of successive lines
+    """
+
+    fst_lineset_index: Index
+    snd_lineset_index: Index
+
+    def __repr__(self) -> str:
+        return (
+            f"<LineSetStartCouple <{self.fst_lineset_index};{self.snd_lineset_index}>>"
+        )
+
+    def __eq__(self, other) -> bool:
+        if not isinstance(other, LineSetStartCouple):
+            return NotImplemented
+        return (
+            self.fst_lineset_index == other.fst_lineset_index
+            and self.snd_lineset_index == other.snd_lineset_index
+        )
+
+    def __hash__(self) -> int:
+        return hash(self.fst_lineset_index) + hash(self.snd_lineset_index)
+
+    def increment(self, value: Index) -> "LineSetStartCouple":
+        return LineSetStartCouple(
+            Index(self.fst_lineset_index + value),
+            Index(self.snd_lineset_index + value),
+        )
+
+
+LinesChunkLimits = Tuple["LineSet", LineNumber, LineNumber]
```

Comment:
```suggestion
LinesChunkLimits_T = Tuple["LineSet", LineNumber, LineNumber]
```
Just to keep it consistent: I would recommend to append `_T` / `T` only to and for all type aliases.

Reply:
Done